core/cog-platform.c

```diff
@@ -138,3 +138,69 @@ cog_platform_create_im_context(CogPlatform *platform)
 
     return NULL;
 }
+
+/**
+ * cog_platform_configure: (constructor)
+ * @name: (nullable): Name of the platform implementation to use.
+ * @params: (nullable): Parameters passed to the platform implementation.
+ * @env_prefix: (nullable): Prefix for environment variables to check.
+ * @shell: The shell that the platform will be configured for.
+ * @error: Location where to store errors, if any.
+ *
+ * Creates and configures a platform implementation.
+ *
+ * Search and create a platform implementation with the given @name,
+ * configuring it with the passed @params for use with a @shell.
+ *
+ * If the @env_prefix is non-%NULL, then the environment variable
+ * `<env_prefix>_PLATFORM_NAME` can be used to set the platform name
+ * when @name is %NULL, and the variable `<env_prefix>_PLATFORM_PARAMS`
+ * can be used to set the configuration parameters when @params is %NULL.
+ * Environment variables will *not* be used if %NULL is passed as the
+ * prefix.
+ *
+ * If both @name is %NULL and the `<env_prefix>_PLATFORM_NAME` variable
+ * not defined, then the platform implementation will be chosen automatically
+ * among the available ones.
+ *
+ * Note that [id@cog_modules_add_directory] may be used beforehand to
+ * configure where to search for available platform plug-ins.
+ *
```

Comment:
It seems this can only be called a single time, maybe that should be documented?

Reply:
It should be possible to call `cog_modules_add_directory` more than once just fine.

Reply:
I meant the configure function. It sets a global and warns if it was already set.